### PR TITLE
Enable storage for shell projects

### PR DESCRIPTION
**Pull request by MatthewBemis**
also see: https://github.com/broadinstitute/workbench-libs/pull/135

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -497,21 +497,21 @@ paths:
...
+      summary: gets a key for the default pet service account for the user
       responses:
         200:
           description: 'json format key for a pet service account, see https://cloud.google.com/iam/docs/creating-managing-service-account-keys'
         500:
           description: Internal Error
           schema:
             $ref: '#/definitions/ErrorReport'
-      operationId: getArbitraryPetServiceAccountKey
+      operationId: getDfaultPetServiceAccountKey
```
Comment (jmthibault79) [inline src/main/resources/swagger/api-docs.yaml]: speling

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 2]
```diff
@@ -281,23 +281,27 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
         case gjre: GoogleJsonResponseException if gjre.getDetails.getCode == StatusCodes.Conflict.intValue => None
       }
       _ <- creationOperationId match {
-        case Some(opId) => pollShellProjectCreation(opId) //poll until it's created
+        case Some(opId) => for {
+          _ <- pollShellProjectCreation(opId) //poll until it's created
+          enableBillingOperationId <- googleServiceManagerDAO.enableService(projectName, "storage-api.googleapis.com")
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: This is not enable billing operation. Does enabling this api require billing?

Comment (MatthewBemis) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: no it doesn't require billing to be enabled. i was trying to get this out quick before leaving while also thinking of something else and the two things merged in my head somehow. so anything that says "enable billing" should be "enable storage apis". 

something along the lines of "do what i mean, not what i say" or whatever that saying is...

**Comment by MatthewBemis** [comment]
jenkins retest because unit tests couldn't connect to opendj during build (see here: https://fc-jenkins.dsp-techops.broadinstitute.org/job/sam-build/897/console)

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 3]
```diff
@@ -281,23 +281,27 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
         case gjre: GoogleJsonResponseException if gjre.getDetails.getCode == StatusCodes.Conflict.intValue => None
       }
       _ <- creationOperationId match {
-        case Some(opId) => pollShellProjectCreation(opId) //poll until it's created
+        case Some(opId) => for {
+          _ <- pollShellProjectCreation(opId) //poll until it's created
+          enableStorageOpId <- googleServiceManagerDAO.enableService(projectName, "storage-api.googleapis.com")
+          _ <- pollServiceOperation(enableStorageOpId) //poll until it's enabled
+        } yield ()
```
Comment (gpolumbo-broad) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala:288]: Nested `for` comprehensions, maybe keep this as an extracted method for readability?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 4]
```diff
@@ -281,23 +281,27 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
...
-    def whenCreating(throwable: Throwable): Boolean = {
-      throwable match {
-        case t: WorkbenchException => throw t
-        case t: Exception => true
-        case _ => false
-      }
+  private def whenOperationActive(throwable: Throwable): Boolean = {
+    throwable match {
+      case t: WorkbenchException => throw t
+      case t: Exception => true
```
Comment (gpolumbo-broad) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala:298]: I'm worried about exceptions being swallowed by this statement.  Can you be more specific on the types of exceptions this is catching.  Maybe it is what you want, but maybe a quick log statement or something that we have an exception that we're not doing anything with.  At the very least, I would at least comment this line justifying why it's OK.

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**